### ISSUE TITLE
Messy audio device ids

### ISSUE THREAD

**Issue by cgarbacea** [livekit/client-sdk-flutter#511]
Can we please do something about the audio ids or add a simple type like builtin | bluetooth | wired, both Android and iOS provie this identification in native (not sure about "wired" :) )

To me it looks like this on Android with some Bluetooth headset attached:
<img width="1036" alt="Screenshot 2024-05-07 at 19 32 06" src="https://github.com/livekit/client-sdk-flutter/assets/133370007/eddd273a-f8c6-49c1-87d2-81ee0a7b6a0b">
On iOS until I set select the  AirPods it looks like this:
<img width="1049" alt="Screenshot 2024-05-07 at 19 33 25" src="https://github.com/livekit/client-sdk-flutter/assets/133370007/54cfa969-d873-4daa-b7de-08aea27692ba">
After I set the AirPods as the input device it looks like this:
<img width="1033" alt="Screenshot 2024-05-07 at 19 35 02" src="https://github.com/livekit/client-sdk-flutter/assets/133370007/7bf8960b-a6f9-4216-a2ee-39e737f401c6">

The bigger issue is the deviceId:
I can't relay on the deviceId or anything to sync the Input/Output, when I switch from one input to another (e.g. select Airpods as the input an i am still on Speaker for the Output)
I need to be able to switch to the bluetooth device when I detect a change with onDeviceChange.stream.listener, but due to the inconsitency in naming - I don't see away to do it

Please let me know if I am missing sommething.

livekit_client: ^2.1.3

**Comment by davidliu** [comment]
@cgarbacea as an intermediate solution to unblock, you can cross reference against the underlying `enumerateDevices()` function:

```
import 'package:flutter_webrtc/flutter_webrtc.dart' as rtc;

var mediaDeviceInfos = await rtc.navigator.mediaDevices.enumerateDevices();
```

Each MediaDeviceInfo object should have a `groupId` field; on iOS, this corresponds to the [AVAudioSession.Port](https://developer.apple.com/documentation/avfaudio/avaudiosession/port) of the output.

**Comment by cgarbacea** [comment]
thanks for the tip, I'll try your suggestion.

**Comment by cgarbacea** [comment]
It won't make a difference the response from rtc is identical. True, there is a groupId but it is inconsistent, "null" where it will matter :)
I belive you just found the root cause - the issue is in "rtc"
we'll need a better mapping from native e.g.:
/// Type of MediaDevice (ex: Bluetooth Audio, Front Camera)
    public let type: MediaDeviceType
    
    `@objc public enum MediaDeviceType: Int, CustomStringConvertible {
    case audioBluetooth
    case audioWiredHeadset
    case audioBuiltInSpeaker
    case audioHandset
    case videoFrontCamera
    case videoBackCamera
    case other

    public var description: String {
        switch self {
        case .audioBluetooth:
            return "audioBluetooth"
        case .audioWiredHeadset:
            return "audioWiredHeadset"
        case .audioBuiltInSpeaker:
            return "audioBuiltInSpeaker"
        case .audioHandset:
            return "audioHandset"
        case .videoFrontCamera:
            return "videoFrontCamera"
        case .videoBackCamera:
            return "videoBackCamera"
        case .other:
            return "other"
        }
    }
}`

and something similar for Android, maybe pulled from android.media.AudioDeviceInfo and android.hardware.camera2.CameraMetadata

**Comment by davidliu** [comment]
Is this a separate issue? [The audio input/output devices should properly set groupId.](https://github.com/flutter-webrtc/flutter-webrtc/blob/8a880156e0a02e9b4cfd1b39157ba94a6b36eaf1/common/darwin/Classes/FlutterRTCMediaStream.m#L630)

Tested it and all the audio related devices gave their port type.

**Comment by cloudwebrtc** [comment]
@cgarbacea This PR should be a valid fix.
https://github.com/flutter-webrtc/flutter-webrtc/pull/1583

**Comment by cgarbacea** [comment]
yeah, very nice - thanks

**Comment by furkanKotic** [comment]
@cloudwebrtc The problem in [the link](https://github.com/livekit/client-sdk-flutter/issues/376) may be the bug on the IOS side. Could you please review the issue?

https://github.com/livekit/client-sdk-flutter/issues/376
